templates/seL4VMParameters.template.c

```diff
@@ -12,24 +13,29 @@
 /*- set linux_address_config = configuration[me.name].get('linux_address_config') -*/
 /*- set linux_image_config = configuration[me.name].get('linux_image_config') -*/
 
+const vm_config_t vm_config = {
+
 /*- if vm_address_config -*/
 
-const unsigned long ram_base = /*? vm_address_config.get('ram_base') ?*/;
-const unsigned long ram_paddr_base = /*? vm_address_config.get('ram_paddr_base') ?*/;
-const unsigned long ram_size = /*? vm_address_config.get('ram_size') ?*/;
-const unsigned long dtb_addr = /*? vm_address_config.get('dtb_addr') ?*/;
-const unsigned long initrd_addr = /*? vm_address_config.get('initrd_addr') ?*/;
+    .ram = {
+        .phys_base = /*? vm_address_config.get('ram_paddr_base') ?*/,
+        .base = /*? vm_address_config.get('ram_base') ?*/,
+        .size = /*? vm_address_config.get('ram_size') ?*/,
```

Comment:
> @hlyytine: Do I understand you comment [#68 (comment)](https://github.com/seL4/camkes-vm/pull/68#issuecomment-1467695535) right that you have code that needs `ram_size`.

Yes, that is correct. We use that to determine whether mapping requests are inside guest RAM. In that case we map 2 MB large pages. I guess that could be upstreamed some day, though.

> My intention here is that that this explicitly describes the fixed ordinary RAM the VMs gets. There might be additional RAM, but this should be described somewhere else then

What do you mean by "ordinary" RAM?

From guest perspective, guest does not even know whether it's physical memory is stage-2 mapped to frames in the way CAmkES-VM does it or if the frames come from a CAmkES dataport -- and because VMM accesses guest memory (to load images and DTB there, among other purposes) through `vm_ram_touch()` to access guest's vspace, VMM does not have any possibility to tell the difference between "ordinary" RAM and something other. To both the guest and the VMM (excluding this mapping code) they are all the same, as ordinary as it gets.

To me it seems the definition of "ordinary" here boils down whether it is something that is done in a traditional manner or in some other way, regardless the fact that to the guest and the VMM cannot tell the difference. And if we have ability to say that a VMM module that provides mappings worth of `ram_size` bytes at guest physical address `ram_base`, we could not care less about how it is done. The only exception is this new `map_one_to_one` which requires that guest physical addresses are the same as real physical addresses. But I am confident we can come up with clever solution.

> or we have add more documentation how this is supposed to work.

Personally I prefer improving code even if it requires minor architectural changes and documentation, so I would vote this.